File: test-suite/handwritten-src/ts/MapRecordTest.ts
import {TestCase, allTests, assertEq, assertTrue} from "./testutils"
import * as test from "../../generated-src/ts/test";

class MapRecordTest extends TestCase {
    m: test.Test_statics;
    constructor(module: test.Test_module_statics) {
        super(module);
        this.m = module.testsuite;
    }

    testCppMapToJavaMap() {
        this.checkJsMap(this.m.TestHelpers.getMap());
    }

    testEmptyCppMapToJsMap() {
        assertEq(0, this.m.TestHelpers.getEmptyMap().size);
    }

    testCppMapListToJsMapList() {
        var jsMapListRecord = this.m.TestHelpers.getMapListRecord();
        var jsMapList = jsMapListRecord.mapList;
        assertEq(1, jsMapList.length);
        this.checkJsMap(jsMapList[0]);
    }

    testJsMapToCppMap() {
        assertTrue(this.m.TestHelpers.checkMap(this.getJsMap()));
    }

    testEmptyJsMapToCppMap() {
        assertTrue(this.m.TestHelpers.checkEmptyMap(new Map()));
    }

    testJsMapListToCppMapList() {
        var jsMapList = [];
        jsMapList.push(this.getJsMap());
        assertTrue(this.m.TestHelpers.checkMapListRecord({mapList: jsMapList}));
    }

    getJsMap() {
        var jsMap = new Map();
        jsMap.set("String1", BigInt(1));
        jsMap.set("String2", BigInt(2));
        jsMap.set("String3", BigInt(3));
        return jsMap;
    }

    checkJsMap(jsMap) {
        assertEq(3, jsMap.size);
        assertEq(jsMap.get("String1"), BigInt(1));
        assertEq(jsMap.get("String2"), BigInt(2));
        assertEq(jsMap.get("String3"), BigInt(3));
    }
}

allTests.push(MapRecordTest);

File: test-suite/handwritten-src/swift/MapRecordTest.swift
import XCTest
@testable import TestSuite

final class MapRecordTest: XCTestCase {
    func testCppMapToSwiftMap() throws {
        checkSwiftMap(try TestHelpers_statics.getMap())
    }

    func testEmptyCppMapToSwiftMap() throws {
        XCTAssertEqual(0, try TestHelpers_statics.getEmptyMap().count)
    }

    func testCppMapListToSwiftMapList() throws {
        let sMapListRecord = try TestHelpers_statics.getMapListRecord()
        let sMapList = sMapListRecord.mapList
        XCTAssertEqual(1, sMapList.count)
        checkSwiftMap(sMapList[0])
    }

    func testSwiftMapToCppMap() throws {
        XCTAssertTrue(try TestHelpers_statics.checkMap(["String1": 1, "String2": 2, "String3": 3]))
    }

    func testEmptySwiftMapToCppMap() throws {
        XCTAssertTrue(try TestHelpers_statics.checkEmptyMap([:]))
    }

    func testSwiftMapListToCppMapList() throws {
        let sMapList: [Dictionary<String, Int64>] = [["String1": 1, "String2": 2, "String3": 3]]
        XCTAssertTrue(try TestHelpers_statics.checkMapListRecord(MapListRecord(mapList:sMapList)))
    }

    func checkSwiftMap(_ smap: Dictionary<String, Int64>) {
        XCTAssertEqual(3, smap.count)
        XCTAssertEqual(smap["String1"], 1)
        XCTAssertEqual(smap["String2"], 2)
        XCTAssertEqual(smap["String3"], 3)
    }
}
